diffpy/snmf/subroutines.py

```diff
@@ -84,6 +84,35 @@ def construct_stretching_matrix(components, number_of_components, number_of_sign
     return stretching_factor_matrix
 
 
+def construct_component_matrix(components, number_of_components, signal_length):
+    """Constructs the component matrix
+
+    Parameters
+    ----------
+    components: tuple of ComponentSignal objects
+      The tuple containing the component signals in ComponentSignal objects.
+    number_of_components: int
+      The number of component signals to obtain from the stretched nmf decomposition.
+    signal_length: int
+      The number of signals in the data provided by the user.
+
+    Returns
+    -------
+    2d array
+      The matrix containing the component signal values. Has dimensions `signal_length` x `number_of_components`.
+
+    """
+    if signal_length <= 0:
+        raise ValueError(f"Signal length = {signal_length}. Signal length must be >= 1")
+    if (number_of_components <= 0) or (len(components) <= 0):
+        raise ValueError(f"Number of components = {number_of_components}. Number_of_components must be >= 1")
+
+    component_matrix = np.zeros((signal_length, number_of_components))
```

Comment:
this is a slightly ugly way of doing it.  Better to maybe use an append.  Also, is there a reason we are not doing like
```
component_matrix[i] = component.iq
```

Reply:
component_matrix is a 2d array so I think that you need two indices. I'm trying to replace each column with columbia.iq

Reply:
This doesn't change that because you you are putting a vector at each position, i.  It is just syntactically cleaner.

Reply:
I see. I will change it